awx/main/utils/analytics_proxy.py

```
@@ -130,16 +137,19 @@ def _generate_access_token(self) -> None:
                 'client_id': self.client_id,
                 'client_secret': self.client_secret,
                 'scope': self.scopes,
             },
-            headers={'Content-Type': 'application/x-www-form-urlencoded'},
-        )
+            'headers': {'Content-Type': 'application/x-www-form-urlencoded'},
+        }
+        proxies = getproxies()
+        if proxies:
+            kwargs['proxies'] = proxies
+        response = requests.post(self.token_url, **kwargs)
```

Ruff (0.15.15)

[error] Probable use of `requests` call without timeout

(S113)